### PR TITLE
Image delete functionality

### PR DESCRIPTION
**Pull request by dodumosu**
this PR implements nditech/apollo-issues#182 and adds front end functionality to delete images.

### CODE REVIEW

File: apollo/frontend/templates/frontend/macros/submission_field_macros.html [hunk 1]
```diff
@@ -11,20 +11,24 @@
...
-<label class="btn btn-light mx-2"><span role="button" tabindex="1"><i class="fas fa-file-image text-primary"></i></span></label>
-<label class="btn btn-danger mx-2"><span role="button" tabindex="1"><i class="fa fa-trash"></i></span></label>
+<label class="btn btn-light mx-2" data-presence-label="{{ form_field.short_name }}"><span role="button" tabindex="1"><i class="fas fa-file-image text-primary"></i></span></label>
+{% if perms.delete_images.can() %}
+<label class="btn btn-danger mx-2" data-delete-label="{{ form_field.short_name }}"><span role="button" tabindex="1"><i class="fa fa-trash"></i></span></label>
 <label class="btn btn-secondary mx-2" style="cursor: pointer;" data-context="select_button">
   <span role="button" tabindex="0"><i class="fas fa-file-upload"></i></span>
   <input type="file" name="{{ form_field.name }}" id="{{ form_field.id }}" data-context="hidden_file" accept=".jpg,.png,image/jpeg,image/png" hidden />
 </label>
+{% endif %}
```
Comment (takinbo) [inline apollo/frontend/templates/frontend/macros/submission_field_macros.html]: I would have thought the permission only affects the delete button but here it also prevents uploading a replacement. Why?

**Review by takinbo — CHANGES_REQUESTED** [review]

**Comment by dodumosu** [comment]
The linked issue specifically mentioned not allowing uploads as well.

**Review by takinbo — CHANGES_REQUESTED** [review]
A few changes requested: 

1. The user should be able to delete an image upload that has not been saved. This means that it would not make a backend request to delete the image since it hasn't been uploaded yet.
2. Add the `cursor: pointer` style option when the delete button can be interacted with. Note that simply disabling the button doesn't prevent the pointer from looking the way it's been specified with the style.

**Comment by takinbo** [comment]
Screenshot from 2022-10-04 18-19-34

Could you get rid of the notification displayed on the screen after deleting an image.

File: apollo/frontend/templates/frontend/submission_edit.html [hunk 2]
```diff
@@ -680,9 +701,69 @@ <h5 class="modal-title">{{ _('Add Call Log') }}</h5>
...
+        payload: response.json()
+      };
+    }).then(function (message) {
+      var galleryItemSelector = 'div[data-gallery-item-tag=' + tag + ']';
+      document.querySelector(deleteLabelSelector).classList.toggle('disabled');
+      document.querySelector(deleteLabelSelector).style.cursor = 'default';
+      imagePresenceLabel.classList.toggle('disabled');
+      imagePresenceLabel.firstChild.firstChild.classList.remove('text-primary');
+      imagePresenceLabel.firstChild.firstChild.classList.add('text-dark');
+      document.querySelector(galleryItemSelector).remove();
```
Comment (takinbo) [inline apollo/frontend/templates/frontend/submission_edit.html]: This assumes that the request will always succeed. Shouldn't it only do this if the response was OK?

**Review by takinbo — CHANGES_REQUESTED** [review]

**Review by takinbo — CHANGES_REQUESTED** [review]
Weird behavior: When an image is selected for the first time, the delete button is enabled (correctly) but when another image is selected (without clicking the delete button), it get disabled which shouldn't be.

**Comment by dodumosu** [comment]
> Weird behavior: When an image is selected for the first time, the delete button is enabled (correctly) but when another image is selected (without clicking the delete button), it get disabled which shouldn't be.

this should be fixed

**Review by takinbo — APPROVED** [review]

**Pull request merged**